train_colors.py

```diff
@@ -0,0 +1,144 @@
+from __future__ import print_function
+
+import os
+import sys
+import numpy as np
+from tqdm import tqdm
+from itertools import chain
+
+import torch 
+import torch.nn as nn
+import torch.optim as optim
+import torch.nn.functional as F
+from torch.utils.data import DataLoader
+from torchvision.utils import save_image
+from color_dataset import (ColorDataset, WeakSup_ColorDataset)
+
+from utils import (AverageMeter, save_checkpoint)
+from models import Supervised
+
+def train(epoch):
```

Comment:
defining this function outside of the `if __name__ == "__main__"` is problematic because variables like `train_loader` are not defined yet. can you move this back into the bottom section